postgres-appliance/callback_role.py

```diff
@@ -8,57 +8,68 @@
 import sys
 import time
 
-TOKEN_FILENAME = '/var/run/secrets/kubernetes.io/serviceaccount/token'
-CA_CERT_FILENAME = '/var/run/secrets/kubernetes.io/serviceaccount/ca.crt'
-API_URL = 'https://kubernetes.default.svc.cluster.local/api/v1/namespaces/{0}/pods/{1}'
+KUBE_SERVICE_DIR = '/var/run/secrets/kubernetes.io/serviceaccount/'
+KUBE_NAMESPACE_FILENAME = KUBE_SERVICE_DIR + 'namespace'
+KUBE_TOKEN_FILENAME = KUBE_SERVICE_DIR + 'token'
+KUBE_CA_CERT = KUBE_SERVICE_DIR + 'ca.crt'
+
+KUBE_API_URL = 'https://kubernetes.default.svc.cluster.local/api/v1/namespaces/{0}/pods/{1}'
 
 logger = logging.getLogger(__name__)
 
 NUM_ATTEMPTS = 10
 LABEL = 'spilo-role'
 
 
-def change_host_role_label(new_role):
+def read_first_line(filename):
     try:
-        with open(TOKEN_FILENAME, "r") as f:
-            token = f.read()
+        with open(filename) as f:
+            return f.readline().rstrip()
     except IOError:
-        sys.exit("Unable to read K8S authorization token")
+        return None
+
+
+def read_token():
+    return read_first_line(KUBE_TOKEN_FILENAME)
 
-    headers = {'Authorization': 'Bearer {0}'.format(token)}
-    headers['Content-Type'] = 'application/json-patch+json'
-    url = API_URL.format(os.environ.get('POD_NAMESPACE', 'default'),
-                         os.environ['HOSTNAME'])
-    data = [{'op': 'add', 'path': '/metadata/labels/{0}'.format(LABEL), 'value': new_role}]
+
+def change_pod_role_label(new_role):
+    pod_namespace = os.environ.get('POD_NAMESPACE', read_first_line(KUBE_NAMESPACE_FILENAME)) or 'default'
+    api_url = KUBE_API_URL.format(pod_namespace, os.environ['HOSTNAME'])
+    body = json.dumps({'metadata': {'labels': {LABEL: new_role}}})
     for i in range(NUM_ATTEMPTS):
         try:
-            r = requests.patch(url, headers=headers, data=json.dumps(data), verify=CA_CERT_FILENAME)
+            token = read_token()
+            if not token:
+                logger.warning('Unable to read K8S authorization token')
+                continue
```

Comment:
Should there be no delay between failed attempts to read the file?  I'd think if it fails, then trying 10 times w/o a delay will fail for sure as well.